src/login/logind-dbus.c

```diff
@@ -2040,17 +2040,23 @@ static int verify_shutdown_creds(
         }
 
         if (blocked) {
-                /* We don't check polkit for root here, because you can't be more privileged than root */
-                if (uid == 0 && (flags & SD_LOGIND_ROOT_CHECK_INHIBITORS))
+                if (!FLAGS_SET(flags, SD_LOGIND_SKIP_INHIBITORS))
                         return sd_bus_error_setf(error, SD_BUS_ERROR_ACCESS_DENIED,
```

Comment:
this should return a proper dbus error btw, that says BlockedByInhibitor... (yes, pre-existing issue i know, but please fix if you touch this)